msamp/operators/fp8_op/fp8_op.py

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Fp8Op module."""
+
+import ctypes
+
+
+class Fp8Op:
+    """Msampfp8 library wrapper class."""
+    lib = ctypes.cdll.LoadLibrary('libmsampfp8.so')
+
+    @classmethod
+    def disable_fp8(self):
+        """Disable fp8. It means uint8/int8 will not be treated as fp8 in ncclAllReduce."""
+        Fp8Op.lib.disable_fp8()
+
+    @classmethod
+    def enable_fp8_e4m3(self):
+        """Enable fp8. It means uint8 will be treated as e4m3-fp8 in ncclAllReduce."""
+        Fp8Op.lib.enable_fp8_e4m3()
+
+    @classmethod
+    def enable_fp8_e5m2(self):
+        """Enable fp8. It means uint8 will be treated as e5m2-fp8 in ncclAllReduce."""
```

Comment:
make e4m3/e5m2 as a parameter in python function?

Reply:
I think there is no need to introduce new enum or types. Current interface is very straightforward.

Reply:
to use fp8 all_reduce(tensor) elsewhere, currently it will always need:
1. get meta.qtype of the tensor
2. check whether the qtype is kfloat8_e4m3, kfloat8_e5m2, or others
3. call enable_fp8_e4m3 or enable_fp8_e5m2 accordingly
4. run all_reduce(tensor)
5. call disable_fp8

I think it should be one interface